packages/flutter_tools/lib/src/base/error_handling_file_system.dart

```diff
@@ -237,33 +255,156 @@ class ErrorHandlingDirectory
+      platform: _platform,
+      failureMessage:
+        'Flutter failed to create a temporary directory with prefix "$prefix"',
+    );
+  }
+
+  @override
+  Directory createTempSync([String prefix]) {
+    return _runSync<Directory>(
+      () => wrap(delegate.createTempSync(prefix)),
+      platform: _platform,
+      failureMessage:
+        'Flutter failed to create a temporary directory with prefix "$prefix"',
+    );
+  }
+
   @override
   String toString() => delegate.toString();
 }
 
 class ErrorHandlingLink
     extends ForwardingFileSystemEntity<Link, io.Link>
     with ForwardingLink {
-  ErrorHandlingLink(this.fileSystem, this.delegate);
+  ErrorHandlingLink({
+    @required Platform platform,
+    @required this.fileSystem,
+    @required this.delegate,
+  }) :
+    assert(platform != null),
+    assert(fileSystem != null),
+    assert(delegate != null),
+    _platform = platform;
 
   @override
   final io.Link delegate;
 
   @override
   final FileSystem fileSystem;
 
+  final Platform _platform;
+
   @override
-  File wrapFile(io.File delegate) =>
-    ErrorHandlingFile(fileSystem, delegate);
+  File wrapFile(io.File delegate) => ErrorHandlingFile(
+    platform: _platform,
+    fileSystem: fileSystem,
+    delegate: delegate,
+  );
 
   @override
-  Directory wrapDirectory(io.Directory delegate) =>
-    ErrorHandlingDirectory(fileSystem, delegate);
+  Directory wrapDirectory(io.Directory delegate) => ErrorHandlingDirectory(
+    platform: _platform,
+    fileSystem: fileSystem,
+    delegate: delegate,
+  );
 
   @override
-  Link wrapLink(io.Link delegate) =>
-    ErrorHandlingLink(fileSystem, delegate);
+  Link wrapLink(io.Link delegate) => ErrorHandlingLink(
+    platform: _platform,
+    fileSystem: fileSystem,
+    delegate: delegate,
+  );
 
   @override
   String toString() => delegate.toString();
 }
+
+Future<T> _run<T>(Future<T> Function() op, {
+  @required Platform platform,
+  String failureMessage,
+}) async {
+  assert(platform != null);
+  try {
+    return await op();
+  } on FileSystemException catch (e) {
+    if (platform.isWindows) {
+      _handleWindowsException(e, failureMessage);
+    } else if (platform.isLinux) {
+      _handleLinuxException(e, failureMessage);
+    }
+    rethrow;
+  }
+}
+
+T _runSync<T>(T Function() op, {
+  @required Platform platform,
+  String failureMessage,
+}) {
+  assert(platform != null);
+  try {
+    return op();
+  } on FileSystemException catch (e) {
+    if (platform.isWindows) {
+      _handleWindowsException(e, failureMessage);
+    } else if (platform.isLinux) {
+      _handleLinuxException(e, failureMessage);
+    }
+    rethrow;
+  }
+}
+
+void _handleLinuxException(FileSystemException e, String message) {
+  // From:
+  // https://github.com/torvalds/linux/blob/master/include/uapi/asm-generic/errno.h
+  // https://github.com/torvalds/linux/blob/master/include/uapi/asm-generic/errno-base.h
+  const int ENOSPC = 28;
```

Comment:
nit: this should follow the naming conventions of the other constants

Reply:
Done